libtock/rubble_ble.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*******************************************************************************
+ *   DRIVER DEFINITIONS   -- corresponds to different system calls
+ *
+ *    *_CMD   - command call
+ *    *_ALLOW - allow call
+ *    *_SUB   - subscribe call
+ *
+ ******************************************************************************/
+
+#define RUBBLE_DRIVER_NUMBER 0x30003
+
+#define RUBBLE_ADV_START_CMD 0
+
+#define RUBBLE_ADV_BUF_ALLOW 0
+
+// start advertising
+//
+// advd               - The advertising data. Must be formatted properly
```

Comment:
Should this be some form of struct rather than a byte array to help guide/enforce this?

Reply:
I'd definitely be interested in adding this in one form or another! But I'm also unsure if it makes sense to right now?

The rubble capsule's minimal interface is mostly in  MVP. I'd like to expose more - maybe even an interface matching rubble's Rust one - but that adds a lot of parsing code that it doesn't otherwise need, so I was thinking it could make more sense as a followup improvement.

The other alternative I know would be to make a struct-based interface in `libtock-c` which simply encodes the response back into bytes. If adding this, I'm not sure if it actually adds any value compared to just using and highly recommending `simple-ble` functions? As I understand it `gap_*` functions should always create valid data, so it seems like we'd just be duplicating that interface. With that said, I'm not a C paradigm expert, so if you think it'd make sense to enforce that with structs at this interface (and possibly just call into `simple-ble`, or duplicate its code?), I'd be down to do that.

Reply:
Mhmm, in the short term, leaving it flexible probably makes the most sense. I'll confess that I don't know the simple_ble stack as well as @brghena -- Branden, is there some form of `ble_adv_data` type that we should consider putting in the interface here, or is just a byte-array appropriate?